packages/skeleton-svelte/src/lib/components/Rating/Rating.svelte

```diff
@@ -1,116 +1,73 @@
+		// Snippets
+		label,
+		// Zag
+		...zagProps
 	}: RatingProps = $props();
 
-	function onRatingMousedown(event: MouseEvent, order: number) {
-		const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
-		const fractionWidth = ratingRect.width / step;
-		const left = event.clientX - ratingRect.left;
-		let selectedFraction = Math.floor(left / fractionWidth) + 1;
-
-		if (getComputedStyle(figureElement).direction === 'rtl') selectedFraction = step - selectedFraction + 1;
-
-		value = order + selectedFraction / step;
-		onmousedown(event, value);
-	}
-
-	// https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio-rating/#kbd_label
-	function onRatingKeydown(event: KeyboardEvent) {
-		const rtl = getComputedStyle(figureElement).direction === 'rtl';
-		if (['ArrowLeft', 'ArrowUp'].includes(event.key)) {
-			event.preventDefault();
-			rtl ? increaseValue() : decreaseValue();
+	// Machine
+	const [state, send] = useMachine(
+		rating.machine({
+			id: useId(),
+			onValueChange: (details) => {
+				value = details.value;
+			}
+		}),
+		{
+			context: {
+				...zagProps,
+				get value() {
+					return value;
+				}
+			}
 		}
-		if (['ArrowRight', 'ArrowDown'].includes(event.key)) {
-			event.preventDefault();
-			rtl ? decreaseValue() : increaseValue();
-		}
-		onkeydown(event);
-	}
-
-	function refreshFocus() {
-		const buttons = figureElement.querySelectorAll('button');
-		// focus on the correct button depending on the new value.
-		buttons[Math.max(0, Math.ceil(value - 1))].focus();
-	}
-
-	function increaseValue() {
-		value = Math.min(max, value + 1 / step);
-		refreshFocus();
-	}
+	);
 
-	function decreaseValue() {
-		value = Math.max(0, value - 1 / step);
-		refreshFocus();
-	}
-
-	let figureElement: HTMLElement;
-
-	// Dynamic Classesclip-path: inset(0 var(--clip_value) 0 0)
-	const rxEmptyInteractive = $derived(interactive ? emptyInteractive : emptyStatic);
-	const rxFullInteractive = $derived(interactive ? fullInteractive : fullStatic);
-
-	// Use the latest rating or 0 if not available, ignoring fractions
-	let focusedButtonIndex = $derived(Math.max(0, Math.ceil(value - 1)));
+	// API
+	const api = $derived(rating.connect(state, send, normalizeProps));
 </script>
 
-<figure class="{base} {width} {justify} {spaceX} {classes}" data-testid="rating" bind:this={figureElement}>
-	{#each { length: max } as _, i}
-		<button
-			class="{buttonBase} {buttonPosition} {buttonAspect} {buttonClasses}"
-			class:pointer-events-none={!interactive}
-			tabindex={interactive && i === focusedButtonIndex ? 0 : -1}
-			onmousedown={(event: MouseEvent) => interactive ? onRatingMousedown(event, i) : undefined}
-			onkeydown={(event: KeyboardEvent) => interactive ? onRatingKeydown(event) : undefined}
-			type="button"
-		>
-			<!-- Icon: Empty -->
-			<span class="{emptyBase} {emptyClip} {rxEmptyInteractive} {emptyClasses}" style="--clipValue: {(value - i) * 100}%">
-				{#if iconEmpty}
+<div {...api.getRootProps()}>
```

Comment:
Make sure you're adding HTML comments within the template to denote each section. This helps match which style props goes to which portion of the template.